From 5c439cdc0adf159c302c626945b1798995a9abe0 Mon Sep 17 00:00:00 2001
From: Robin Murphy <robin.murphy@arm.com>
Date: Thu, 17 Sep 2015 14:52:53 +0100
Subject: [PATCH 27/57] add MSIs to dma debug

Signed-off-by: Robin Murphy <robin.murphy@arm.com>
---
 drivers/iommu/dma-iommu.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/iommu/dma-iommu.c b/drivers/iommu/dma-iommu.c
index c0430d7..3e0b2fd 100644
--- a/drivers/iommu/dma-iommu.c
+++ b/drivers/iommu/dma-iommu.c
@@ -78,13 +78,17 @@ static int iommu_dma_debug_show(struct seq_file *m, void *p)
 		dev = iova->client;
 		if (dev) {
 			phys_addr_t phys = iommu_iova_to_phys(dom, pfn_lo << shift);
+			bool msi = (unsigned long)dev & 1;
 
-			seq_printf(m, "%08lx - %08lx (%lx): %p (%s)\n",
+			dev = (void *)((unsigned long)dev & ~1UL);
+
+			seq_printf(m, "%08lx - %08lx (%lx): %p (%s%s)\n",
 					pfn_lo << shift,
 					(pfn_hi << shift) - 1,
 					(pfn_hi - pfn_lo) << shift,
 					phys_to_virt(phys),
-					dev_name(dev));
+					dev_name(dev),
+					msi ? " [MSI]" : "");
 		} else if (pfn_hi < iovad->dma_32bit_pfn) {
 			seq_printf(m, "%08lx - %08lx reserved\n",
 					pfn_lo << shift,
@@ -655,7 +659,7 @@ int iommu_dma_map_msi_msg(struct msi_desc *desc, struct msi_msg *msg)
 		return 0;
 
 	msi_addr = (phys_addr_t)((u64)msg->address_hi << 32 | msg->address_lo);
-	iova = __alloc_iova(iovad, 1, dma_get_mask(dev), dev);
+	iova = __alloc_iova(iovad, 1, dma_get_mask(dev), (void *)((unsigned long)dev | 1));
 	if (!iova)
 		return -ENOSPC;
 
-- 
2.7.4

